File: 125. Valid Palindrome/isPalindrome.py
class Solution:
  def isPalindrome(self, s: str) -> bool:
    left, right = 0, len(s) - 1
    while left < right:
      if s[left].isalnum() == False:
        left += 1
        continue

      if s[right].isalnum() == False:
        right -= 1
        continue

      if s[left].lower() != s[right].lower():
        return False
      
      left += 1
      right -= 1

    return True


s = "A man, a plan, a canal: Panama"
solution = Solution()
print('isPalindrome: ', solution.isPalindrome(s)) # True

File: 125. Valid Palindrome/isPalindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
const isPalindrome = (s) => {
  let left = 0;
  let right = s.length - 1;
  const isAlphaNumeric = (char) => {
    return /[a-zA-Z0-9]/.test(char);
  };

  while (left < right) {
    if (!isAlphaNumeric(s[left])) {
      left++;
      continue;
    }

    if (!isAlphaNumeric(s[right])) {
      right--;
      continue;
    }

    if (s[left].toLowerCase() !== s[right].toLowerCase()) {
      return false;
    }

    left++;
    right--;
  }

  return true;
};


const s = "A man, a plan, a canal: Panama";
console.log('isPalindrome: ', isPalindrome(s)); // true
